src/main/java/jenkins/metrics/impl/JenkinsMetricProviderImpl.java

```diff
@@ -773,11 +825,33 @@ public boolean shouldSchedule(Queue.Task p, List<Action> actions) {
         private final Map<Queue.BlockedItem, Timer.Context> blocked = new WeakHashMap<>();
         private final Map<Queue.BuildableItem, Timer.Context> buildable = new WeakHashMap<>();
         private final Map<Queue.WaitingItem, Timer.Context> waiting = new WeakHashMap<>();
+        private final AtomicLong nextTrim = new AtomicLong(System.nanoTime());
+        private final ConcurrentMap<Long, ItemTotals> totals = new ConcurrentHashMap<>();
+        private Set<Long> previousIds = new HashSet<>();
 
         public static ScheduledRate instance() {
             return ExtensionList.lookup(QueueListener.class).get(ScheduledRate.class);
         }
 
+        private void trim() {
+            long next = nextTrim.get();
+            if (next - System.nanoTime() < 0L && nextTrim.compareAndSet(next, next + TRIM_INTERVAL_NANOS)) {
+                // we won the trim lottery
+                Queue queue = Jenkins.getInstance().getQueue();
+                // get the IDs that are currently known
+                Set<Long> currentIds = Stream.concat(Stream.of(queue.getItems()), queue.getLeftItems().stream())
+                        .map(Queue.Item::getId)
+                        .collect(Collectors.toCollection(HashSet::new));
+                // remove the currently known IDs from the previous list of known IDs
+                previousIds.removeAll(currentIds);
+                // now remove from the totals the ones that were previously known (at last trim) but are
+                // no longer known, we need to do it this was as otherwise any IDs that are added while
```

Comment:
There is confusing comment grammar here. Also possibly the comment is incomplete?